### DISCUSSION TITLE
Did "vanished file" affect other file transfers？

### DISCUSSION THREAD

**Discussion by qqliuxiaoran** [RsyncProject/rsync#485] [Q&A] [answered]
Hello WayneD and team!
I want to know if the warning of 'rsync warning: some files vanished before they could be transferred (code 24)' affect the transfer of other files in same batch?  I noticed your explanation of the exit code on the official website: **24 -⁠ Partial transfer due to vanished source files**, but I can't get an accurate meaning from this sentence.  When I encounter this exit code, should I try again or just ignore it？
Thank you and look forward your reply!

**Comment by michaelmess-de** [comment]
I have deleted a few source files while a transfer was in progress. 
Some files were reported as vanished, but the transfer continued.
At the end there was a rsync warning: some files vanished before they could be transferred (code 24).
Restarting rsync didn't bring any new files to transfer.

Thus if you didn't add any new files in the meantime that should be transfered, you can ignore it.
Of course the vanished files have not been transfered and didn't make it to the target, but all other files should have made it to the target.
But if you have added any new files in the meantime, then you might want to restart rsync regardless of any vanished files.

Rsync collects a list of files to be transfered at start time, then starts the transmission.
Thus any files contained in the list but not existing anymore are reported as vanished.
Vanished files do not break the running transmission.
Any new files that didn' t exist when collecting the list are silently ignored during the running transfer.

**Comment by steadytao** [comment, marked as answer]
Pretty much entirely correct. Par the fact that rsync normally uses incremental recursion nowadays. But repeating from Michael.

Exit code 24 does not stop the remaining files from being transferred. It means one or more source files disappeared after rsync discovered them but before it could transfer them so those files are missing from the destination.

Whether you should retry depends on what you are copying. For a live tree where temporary files are regularly created and removed, inspect the reported paths and ignore the warning if those files do not matter. If you need a complete and consistent backup, rerun the transfer and consider taking a snapshot or otherwise preventing the source from changing during the run.
